src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinSourceEdgeWriter.java

```diff
@@ -50,11 +44,11 @@ public void write(Object domain, MappingGremlinConverter converter, GremlinSourc
 
             if (field.getName().equals(Constants.PROPERTY_ID)) {
                 continue;
-            } else if (field.getAnnotatedType().getType() == EdgeFrom.class) {
-                sourceEdge.setVertexIdFrom(this.getPersistentEntityId(object.getClass()));
+            } else if (field.getAnnotation(EdgeFrom.class) != null) {
+                sourceEdge.setVertexIdFrom(super.getEntityIdValue(object, converter));
                 continue;
-            } else if (field.getAnnotatedType().getType() == EdgeTo.class) {
-                sourceEdge.setVertexIdTo(this.getPersistentEntityId(object.getClass()));
+            } else if (field.getAnnotation(EdgeTo.class) != null) {
```

Comment:
so edgefrom and edgeto cannot co-exists in domain class?

Reply:
no, edgefrom and edgeto may act on different field in class, here only iterate the field to find them. I think we may add some checking here for more than one edgeFrom/To annotation in domain class, will refine this later.

Reply:
file one issue #1 for this